src/main/java/org/codehaus/plexus/util/xml/pull/MXParser.java

```diff
@@ -2664,7 +2664,8 @@ else if ( ch == '\t' || ch == '\n' || ch == '\r' )
         entityRefName = null;
         posStart = pos;
         char ch = more();
-        StringBuilder sb = new StringBuilder();
+        StringBuilder sb16 = new StringBuilder();
+        StringBuilder sb10 = new StringBuilder();
```

Comment:
instead of creating 2 separate string buffers, keeping sb and creating a "isHex" boolean would be a little bit more efficient and permit code sharing between the 2 cases since it's only a question of radix value, isn't it?